### ISSUE TITLE
multi-line fig-cap in .qmd notebook still broken

### ISSUE THREAD

**Issue by rmnldwg** [quarto-dev/quarto-cli#9421]
### Bug description

First, I think this is the same issue as #6521. But it not only affects the rendering of `.qmd` files to `.ipynb`, but also the notebooks of manuscript projects. I hope it was ok to open a new issue, I did not know if a comment under the old one would have been enough or not.

When using the "manuscript" project, long `fig-cap`s cause comically wide HTML previews. Trying to use a multi-line `fig-cap` does not work at all: It gets wrongly transformed and no figure output is displayed.

### Steps to reproduce

First, the `_quarto.yml` file:
```yaml
project:
  type: manuscript

manuscript:
  article: index.qmd
  notebooks: 
    - notebook: figures.qmd

format:
  html: default
```

Then the main manuscript `index.qmd`:
````qmd
---
title: Manuscript
authors:
  - name: Roman Ludwig
---

## Section
This is a simple placeholder for the manuscript's main document.

{{< embed figures.qmd#fig-some-lines >}}
````

The notebook `figures.qmd`, once with one long `fig-cap` line:
````qmd
---
title: Plotting Figures
---
```{python}
#| label: fig-some-lines
#| fig-cap: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Donec ultrices tincidunt arcu non. Nulla aliquet porttitor lacus luctus accumsan tortor posuere ac. Dui accumsan sit amet nulla facilisi. Et tortor at risus viverra.
import matplotlib.pyplot as plt
import numpy as np

x = np.linspace(0., 1., 10)
data = np.random.uniform(size=10)
plt.plot(x, data)

plt.show()
```
````

And then lastly, the `figures.qmd` notebook with a multi-line `fig-cap`:
````qmd
---
title: Plotting Figures
---
```{python}
#| label: fig-some-lines
#| fig-cap: >-
#|   Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
#|   incididunt ut labore et dolore magna aliqua. Donec ultrices tincidunt arcu non.
#|   Nulla aliquet porttitor lacus luctus accumsan tortor posuere ac. Dui accumsan sit
#|   amet nulla facilisi. Et tortor at risus viverra.
import matplotlib.pyplot as plt
import numpy as np

x = np.linspace(0., 1., 10)
data = np.random.uniform(size=10)
plt.plot(x, data)

plt.show()
```
````

### Expected behavior

A single long line on a notebook's code cell should not cause the entire document to get rendered as wide as that cell, I guess?

But more importantly, multi-line `fig-cap`s should work as one would expect YAML to work.

### Actual behavior

In the first case, when using one long line in the `fig-cap`, the resulting notebook gets rendered way too wide:

![too wide](https://github.com/quarto-dev/quarto-cli/assets/48687784/54b2e50b-abe1-4b94-8ced-a78432760802)

While in the second case, it does not get rendered at all, because the figure caption is wrongly formatted (I presume):

![not diplayed](https://github.com/quarto-dev/quarto-cli/assets/48687784/f411d98b-4e51-41f5-9d18-2fe7842ff6ce)

Somehow it transforms the `fig-cap: >-` line to `fig-cap: |-` and inserts a blank line afterwards.

Note that in the main article, it looks fine when using the single long line, since there the code cell is not displayed.

### Your environment

- IDE: VS Code
- OS: Ubuntu 22.04 inside WSL2 on Widnows 11

I installed `numpy`, `matplotlib`, and `jupyter` to create this minimal plot.

### Quarto check output

```bash
Quarto 1.5.30
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.13: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.30
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.04
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/rmnldwg/.TinyTeX/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /home/rmnldwg/repos/quarto-test/.venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)
```

**Comment by mcanouil** [comment]
Sorry, but is all of that needed to reproduce?
Manuscript, the embed and the first `figures.qmd` does not seem to be useful.

The following seems quite enough:

````qmd
---
title: Plotting Figures
---

```{python}
#| label: fig-some-lines
#| fig-cap: >-
#|   Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
#|   incididunt ut labore et dolore magna aliqua. Donec ultrices tincidunt arcu non.
#|   Nulla aliquet porttitor lacus luctus accumsan tortor posuere ac. Dui accumsan sit
#|   amet nulla facilisi. Et tortor at risus viverra.
import matplotlib.pyplot as plt
import numpy as np

x = np.linspace(0., 1., 10)
data = np.random.uniform(size=10)
plt.plot(x, data)

plt.show()
```
````

**Comment by rmnldwg** [comment]
Thanks for the quick response and sorry for the convoluted example!

I only encountered this issue in the supporting notebooks of the `manuscript` project type. Because in a `default` project (or in the main article of the `manuscript` project)...

1. ... the cell tags are not rendered, thus the long `fig-cap` doesn't bother me.
2. ... the rendered HTML code cell has a horizontal scroll bar (see image below).

![image](https://github.com/quarto-dev/quarto-cli/assets/48687784/f0aad0ab-9cea-4c7d-a596-d189ab4dbbc3)

So, I guess I wanted to show that it's sort of two related problems:

1. the wrong handling of cell tag YAML
2. the way code cells are displayed in `manuscript` notebooks

Maybe I should have just opened two issues? Anyways, apart from this minor hiccup, I really do enjoy writing papers with Quarto. Much better than using LaTeX directly! 👍🏻

**Comment by mcanouil** [comment]
> the rendered HTML code cell has a horizontal scroll bar (see image below).

This is expected and can be controlled, see <https://quarto.org/docs/output-formats/html-code.html#code-overflow>.

> the cell tags are not rendered, thus the long fig-cap doesn't bother me.

Sorry, but I don't understand what you mean here.

Note that https://github.com/quarto-dev/quarto-cli/issues/9421#issuecomment-2067363458 shows how to easily reproduce the issue which is unrelated to Manuscript.

**Comment by rmnldwg** [comment]
Sorry for the late response!

Ok, maybe I did a bad job at trying to convey my problem. The issue I was trying to raise only affects the notebooks *supporting* a manuscript article. Like the `data-screening.ipynb` in the [La Palma Earthquakes example](https://quarto-ext.github.io/manuscript-template-jupyter/) from your documentation.

> This [code overflow] is expected and can be controlled, see https://quarto.org/docs/output-formats/html-code.html#code-overflow.

For some reason, in those supporting notebooks to a manuscript project, the setting `code-overflow` does not have any effect (neither in the front matter, nor as a cell tag). This is why my supporting notebooks all become comically wide.

So, basically my issue is purely cosmetical. And it would be solved by either of two things:
1. multi-line cell tag YAML gets parsed correctly
2. `code-overflow` adjustable in notebooks supporting a manuscript project

If the second point is intended, then I am happy to shorten my example above or close the issue completely in favour of reopening #6521.

Thanks for your patience with me!
